src/Zope2/utilities/tests/test_zconsole.py

```diff
@@ -0,0 +1,79 @@
+import os
+import sys
+import tempfile
+import unittest
+
+if sys.version_info.major == 2:
+    from StringIO import StringIO
+else:
+    from io import StringIO
+
+zope_conf_template = """
+%define INSTANCE {}
+
+instancehome $INSTANCE
+
+<zodb_db main>
+   <filestorage>
+     path $INSTANCE/Data.fs
+   </filestorage>
+   mount-point /
+</zodb_db>
+"""
+
+test_script = """
+import sys
+
+
+if __name__ == '__main__':
+    app.foo = '42'
+    print(app.foo)
+    print(sys.argv[1:])
+"""
+
+
+class ZConsoleTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.instancedir = tempfile.mkdtemp(prefix='instance')
+        self.zopeconf = os.path.join(self.instancedir, 'zope.conf')
+        with open(self.zopeconf, 'w') as conffile:
+            conffile.write(zope_conf_template.format(self.instancedir))
+
+    def test_debug(self):
+        stored_sys_argv = sys.argv
```

Comment:
Maybe this can be done in `setUp` and torn down in `tearDown` as each test in this class needs it.